README.md

```diff
@@ -1,21 +1,64 @@
 [![Build Status](http://ci.bazel.io/buildStatus/icon?job=BUILD_file_generator)](http://ci.bazel.io/job/BUILD_file_generator)
 
-# BUILD File Generator
+# BUILD File Generator (BFG)
 
-BUILD File Generator generates Bazel BUILD files for Java code.
+BUILD File Generator, or BFG for short, is a tool for generating Bazel BUILD files from one's source code. Instead of manually going through your project and creating BUILD files and their appropriate build rules, you can rely on BFG to do it for you! 
 
-1. It reads all `.java` files, and extracts the class dependency graph.
-2. Computes the strongly connected components of the graph.
-3. For each component, creates a `java_library` rule.
+An example project can be found here. TODO
+
+### Why is it useful?
+
+Bazel promises fast and correct builds, _especially_ for incremental builds. Rather than rebuild your entire codebase, Bazel _only_ rebuilds what is necessary, the targets that you have changed. For the remainder of your code, it relies on cached versions. 
+
+There is one caveat to this fast incremental performance, though. It is dependent on the granularity of your targets. At one extreme, the least granular end, you can define a single glob for your entire project. This would be the quickest way to get your project using Bazel. However, it comes at the cost of eradicating any of Bazel's incremental build performance. On the other end, you can try to manually define the most granular targets possible, a single rule per file, collapsing any cyclic dependencies into the same target. This would ensure you make use of Bazel's blazingly fast incremental builds. However, it is extremely tedious, and somewhat annoying to maintain.
+
+BFG automates this difficult tango. 
 
 ## Usage
 
-TODO
+BFG is composed of two general components 
+
+1. Language specific parsers (LSP) 
+2. The BFG binary
+
+The LSPs read your source code and generate a class dependency graph in the form of a protobuff. To generate your BUILD files, you pass the generated protobuff into the BFG binary. 
+
+### Step 0: Installation Instructions
+
+To install BFG...TODO
```

Comment:
Do we have an eventual goal for users will download and install BFG? Would it go in [`bazel-dev`](https://github.com/bazelbuild/bazel/issues/3895)?

Reply:
As far as I know, we have yet to define that goal. 

I think it would be nice if users didn't have to install BFG at all, and they could something like 
```
bazel generate <options-and-configs> project/
```
What do other projects do on this front? I am not the hugest fan of needing to `go get` buildozer.